### ISSUE TITLE
Project idea: Perl 6 Testing Done Right

### ISSUE THREAD

**Issue by AlexDaniel** [perl-gsoc-2019/ideas#9]
## Description

Writing tests in perl 6 is typically done using the core `Test` module. Some of its functionality is less than awesome and error-prone, which prompted people to write alternatives (see [Testo](https://github.com/perl6-community-modules/perl6-Testo)).

## Expected outcomes

* Review the alternatives and decide what are the biggest issues with routines provided by `use Test`.
* Decide what can and should be changed.
* Implement these changes in a versioned manner (so that test files that `use v6.c` or `use v6.d` remain working, and the changes are only affecting files that use v6.e language version).

## Required skills

Experience with Perl 6 or desire to learn Perl 6.

## Rating

Medium

## Possible mentors

Write your name here please.

**Comment by patzim** [comment]
I'd say this needs good mentoring during the research phase. Also experience with testing on the student side is a nice-to have I'd say.

**Comment by AlexDaniel** [comment]
@Leont can be a great mentor for this project: https://colabti.org/irclogger/irclogger_log/perl6?date=2019-01-28#l404

**Comment by AlexDaniel** [comment]
Oh, anybody else should also feel free to mentor this. @patzim if you feel like doing it then please do.

**Comment by Leont** [comment]
Yeah, I think I have a good idea of what's we would want from such a module, and roughly how to get there.

Can't promise not to write such a thing myself before summer though ;-)

**Comment by JJ** [comment]
Thanks, Leon. I just created another issue in Rakudo, just in case
https://github.com/rakudo/rakudo/issues/2656. I can help you with this,
tutoring, or whatever you need.
